### DISCUSSION TITLE
cffwis_indices: warnings and errors

### DISCUSSION THREAD

**Discussion by GvdDool** [Ouranosinc/xclim#1934] [Questions]
### Setup Information

* Xclim version: '0.47.0'

* windows
* python notebook

### Context

Hi,
I am running the default settings of cffwis_indices, and the data provided:
ERA5: \xclim\daily_surface_cancities_1990-1993.nc

but it seems that the code is very unstable, I managed to run the example once without errors, but rerunning the code gives me the following:

> ...\Lib\site-packages\xclim\core\cfchecks.py:41: UserWarning: Variable does not have a `cell_methods` attribute.
>   _check_cell_methods(
> ...\Lib\site-packages\xclim\core\cfchecks.py:45: UserWarning: Variable does not have a `standard_name` attribute.
>   check_valid(vardata, "standard_name", data["standard_name"])

on this code:
```
season_mask = xc.indicators.atmos.fire_season(
    tas=ds.tas,
    method="WF93",
    freq="YS",
    # Parameters below are at their default values, but listed here for explicitness.
    temp_start_thresh="12 degC",
    temp_end_thresh="5 degC",
    temp_condition_days=3,
)

```

I know these are not critical, but they are strange when running a test/demo configuration. 

The next errors are more serious:

>...\Lib\site-packages\xarray\core\accessor_dt.py:72: FutureWarning: Index.ravel returning ndarray is deprecated; in a future version this will return a view on self.
>   values_as_series = pd.Series(values.ravel(), copy=False)
> ...\Lib\site-packages\numba\np\ufunc\dufunc.py:202: RuntimeWarning: invalid value encountered in _fine_fuel_moisture_code
>   return super().__call__(*args, **kws)

using the standard function:
```
out_fwi = xc.indices.cffwis_indices(
    tas=ds.tas,
    pr=ds.pr,
    hurs=ds.hurs,
    sfcWind=ds.sfcWind,
    lat=ds.lat,
    season_mask=season_mask,
    overwintering=True,
    dry_start="CFS",
    prec_thresh="1.5 mm/d",
    dmc_dry_factor=1.2,
    # Parameters below are at their default values, but listed here for explicitness.
    carry_over_fraction=0.75,
    wetting_efficiency_fraction=0.75,
    dc_start=15,
    dmc_start=6,
    ffmc_start=85,
)
```

I am not sure what I am doing wrong, as I am loading defaults without changing the variables or parameters. I am trying to reproduce the demo on other data and again running into similar warnings and errors on that data as well, so I would like to understand the demo data first before trying something else. 

Thanks for helping me out here.
Gijs 

### Steps To Reproduce

```
$ pip install foo --bar
```
see code in the other box, using: https://xclim.readthedocs.io/en/stable/apidoc/xclim.indices.fire.html

**Comment by aulemahal** [comment]
Hi @GvdDool,

Nothing seems really problematic in what you showed.

The first two warnings (`Variable does not have a [...] attribute`) is xclim being (overly) cautious about CF standards.  However, these two attribute exist on the test data you are using, so the warning shouldn't appear. Digging a bit more I found out that this is a bug... The CF compliance checker is looking for attributes on `snd` even though you didn't pass one (which is ok). Thanks for noting, this, we'll have to fix that and make those warning disappear.

The other two are also not problematic.

- `[...]xarray\core\accessor_dt.py:72: FutureWarning: Index.ravel [...]` comes from pandas and concerns code in xarray. It is warning you that a deprecated method is being used. The issue lies in xarray and it seems it has been fixed in the latest version. But no need to worry or to upgrade your environment only for this small warning. 

- `numba\np\ufunc\dufunc.py:202: RuntimeWarning: invalid value encountered in _fine_fuel_moisture_code`. That one usually means that somewhere in the computation a `nan` or an `inf` was seen where it was not expected. It might also come from a division by zero.  This happens a lot in real-world situations and usually it will yield a NaN in the output. I wouldn't be worried unless I have a too many NaNs where I thought there would be real values.

(Sorry I didn't mean to close the discussion before you confirm this answer is appropriate!)

**Reply by GvdDool** [reply]
Thanks Pascal Bourgault (https://github.com/aulemahal),
I fixed the first:
```
ds['tas'] = ds['tas'].assign_attrs(
        cell_methods="Time: mean within days",
        long_name="Mean daily surface temperature",
        standard_name= "air_temperature",
        units="C", description="Temperature data")
```
The method didn't like the degree symbol before the C

The ravel, I understand as well, and it's only a warning, so I guess this will be (as you said) be fixed with a future update, in xarray. The last point: __fine_fuel_moisture_code_  is worrying a bit, because in the "real" data I am getting also errors in the later parts, and I don't seem to have a good handle on adjusting the values - perhaps running the methods separately, and filling a very small value for the division by zero?

**Reply by aulemahal** [reply]
Hmm, is the output suspicious ?
Before the beginning of the season, it's all NaN, that might be the reason for this warning ?

Also, if you take the `ds` directly from xclim's testdata, you shouldn't need to plug in attributes, they are already there. Is there something that stripped them away in your code ? 

`units` can be "degC" or "°C". "C" would be interpreted as Coulomb!

**Reply by GvdDool** [reply]
units can be "degC" or "°C". "C" would be interpreted as Coulomb! 

This is strange:
I am working in a notebook, and if I set this:
ds['tas'] = ds['tas'].assign_attrs(
        units="  ", description="Temperature data")
        
and then run:
```
season_mask = xc.indicators.atmos.fire_season(
    tas=ds.tas,
    method="WF93",
    freq="YS",
    # Parameters below are at their default values, but listed here for explicitness.
    temp_start_thresh="12 degC",
    temp_end_thresh="5 degC",
    temp_condition_days=3,
)
```
that generates an error on [temperature] as to be expected, but then updating to:
ds['tas'] = ds['tas'].assign_attrs(
        units="C ", description="Temperature data")
        
the ds is updated and rerunning: season_mask = xc.indicators.atmos.fire_season(...) doesn't generate the warning.

I don't like warnings, so I am trying to remove these when running code, that's how I discovered this bug(?)

**Reply by GvdDool** [reply]
Running the next call:
on real data give this:

> ```
> ..\Lib\site-packages\xarray\core\accessor_dt.py:72: FutureWarning: Index.ravel returning ndarray is deprecated; in a future version this will return a view on self.
>   values_as_series = pd.Series(values.ravel(), copy=False)
> ..\Lib\site-packages\numba\np\ufunc\dufunc.py:202: RuntimeWarning: invalid value encountered in _drought_code
>   return super().__call__(*args, **kws)
> ..\Lib\site-packages\numba\np\ufunc\dufunc.py:202: RuntimeWarning: invalid value encountered in _drought_code
>   return super().__call__(*args, **kws)
> ..\Lib\site-packages\numba\np\ufunc\dufunc.py:202: RuntimeWarning: invalid value encountered in _fine_fuel_moisture_code
>   return super().__call__(*args, **kws)
> ..\Lib\site-packages\numba\np\ufunc\dufunc.py:202: RuntimeWarning: invalid value encountered in _duff_moisture_code
>   return super().__call__(*args, **kws)
> ```

and the code terminates without results, this is over a 21 year time slice, reducing the period gives results, but as we are looking at the whole period it would be nice to run all the years (to avoid having to think about seasonality)

**Reply by GvdDool** [reply]
this is the input data:
//Apply the filter with the dates
dataset2 = ERA5_LAND_DAILY.filter(ee.Filter.date(start_date, end_date)).select('total_precipitation_max', 
                                                                               'dewpoint_temperature_2m_min',
                                                                               'dewpoint_temperature_2m_max',
                                                                               'temperature_2m_min',
                                                                               'temperature_2m_max',
                                                                               'u_component_of_wind_10m_min',
                                                                               'u_component_of_wind_10m_max',
                                                                               'v_component_of_wind_10m_min',
                                                                               'v_component_of_wind_10m_max')

and I am using these variables to construct the same data structure as in the demo.

**Comment by GvdDool** [comment]
The code is working as intended, but I had to implement the analysis by season (year), starting in the low period (January) and stopping in December); there is no overwintering in this area, so that's fine. 

Thanks for your support! 

Steps taken:
1. added the snow depth as a variable (even when not used)
2. suppressed the future warning
3. implemented a run time warning counter (the code encounters 8500 times an error in 44 unique instances)
